lib/matplotlib/dates.py

```diff
@@ -1254,6 +1241,8 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
+        if not isinstance(interval, int) or interval < 1:
```

Comment:
That is a fair point, but for a day locator, does that really make sense?  I personally think it would be better to enforce the typing in this specific case.

Unless you think it would be a problem with existing code bases, then I suppose catering to compatibility would be preferred.